flang/lib/Optimizer/Builder/FIRBuilder.cpp

```diff
@@ -286,6 +288,30 @@ mlir::Value fir::FirOpBuilder::convertWithSemantics(mlir::Location loc,
     auto rp = helper.extractComplexPart(val, /*isImagPart=*/false);
     return createConvert(loc, toTy, rp);
   }
+  if (callingImplicitInterface) {
+    // When passing arguments to a procedure with an implicit interface, allow
+    // character actual arguments to be passed to dummy arguments of any type
+    // and vice versa
```

Comment:
I would move this comment in call lowering (see my comment about `callingImplicitInterface` argument name)